### DISCUSSION TITLE
npm vs yarn

### DISCUSSION THREAD

**Discussion by bogdansoare** [nrwl/nx#6379] [General]
Usually I use `yarn` in my projects, with `Nx` I see that the default recommendation is `npm`.
Are there any downsides in using `yarn` with Nx? Or is `npm` better suited for this kind of monorepo setups?

**Comment by Himadu2000** [comment]
I use yarn in my Nx I had no issue with it. I can't use npm not only it takes heavy load on my network, It takes more than hour to install dependencies. I actually hope Nx default switch over to yarn. And must give a option to create workspace in yarn and pnp support.

**Comment by marckassay** [comment]
I would say your question depends on what version of yarn you're using. I assume if you're using yarn v1, there should be no issues. Offically Nx doesn't support yarn v2, as July 29th:

> Could you try using `yarn` v1 please? Nx does not yet support yarn v2. - [ref](https://github.com/nrwl/nx/issues/6509#issuecomment-889439563)

But I've been using yarn v2, with limited features (No yarn PnP, no Nx migrate), having no issues. @KyleBastien listed a workaround for using yarn v2 and Nx [here](https://github.com/nrwl/nx/issues/2386#issuecomment-890019994).
